### PR TITLE
D-Pad button mappings

### PR DESCRIPTION
**Pull request by jshowacre**
## Overview
- Added individual D-Pad button mappings

This way a developer could add actions for each button, not just the D-Pad

### CODE REVIEW

File: Assets/MixedRealityToolkit/_Core/Definitions/Utilities/AxisType.cs [hunk 1]
```diff
@@ -26,6 +26,10 @@ public enum AxisType
         /// </summary>
         SingleAxis,
         /// <summary>
+        /// Negative Single Axis analogue input.
+        /// </summary>
+        NegativeSingleAxis,
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit/_Core/Definitions/Utilities/AxisType.cs:31]: We already handle this case. and doesn't need to be added.

**Comment by StephenHodgson** [comment]
We already handle the negative axis data. We don't need to do this.

Take a look at: 
https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/2568/files#diff-3598f25fe48b9276349443aeaf3990e6R342

**Comment by jshowacre** [comment]
Unrelated to the inverting the axis, I can do that a better way. What do you think about having the individual D-Pad directions mapped?  Looking something like this:
```
new MixedRealityInteractionMapping(3, "D-Pad Up", AxisType.SingleAxis, DeviceInputType.ButtonPress, ControllerMappingLibrary.AXIS_6)
new MixedRealityInteractionMapping(4, "D-Pad Down", AxisType.NegativeSingleAxis, DeviceInputType.ButtonPress, ControllerMappingLibrary.AXIS_6)
new MixedRealityInteractionMapping(5, "D-Pad Right", AxisType.SingleAxis, DeviceInputType.ButtonPress, ControllerMappingLibrary.AXIS_5)
new MixedRealityInteractionMapping(6, "D-Pad Left", AxisType.NegativeSingleAxis, DeviceInputType.ButtonPress, ControllerMappingLibrary.AXIS_5)
```

imo I think these having actions rather than the whole D-Pad is more intuitive.

**Comment by StephenHodgson** [comment]
We don't have to map them individually.

**Comment by StephenHodgson** [comment]
Ah, I see you're asking to be able to make each dpad button a specific action?

**Comment by jshowacre** [comment]
Yes, I think it'd be useful :)

**Comment by StephenHodgson** [comment]
Ah, so you want them to behave more like buttons.

Let me think about it a bit. I think the way it's setup currently you can do that, but I do see the appeal of having specific actions, similar to how we process the single axis data for determining grip press on WMR controller for OVR.

**Comment by jshowacre** [comment]
Of course, and in my latest commit all I did was add a invertXAxis and it looks a lot better than my first attempt

**Comment by StephenHodgson** [comment]
I'm really happy you made the attempt :)

Doing stuff like this collaboratively helps us both learn.

**Pull request closed without merging**